src/bms/command/channel.rs

```diff
@@ -183,107 +178,194 @@ pub fn read_channel_beat(channel: &str) -> Option<Channel> {
     Some(Channel::Note { kind, side, key })
 }
 
-/// Reads a key from a character. (For PMS)
-fn get_key_pms_bme_type(key: char) -> Option<Key> {
-    use Key::*;
-    Some(match key {
-        '1' => Key1,
-        '2' => Key2,
-        '3' => Key3,
-        '4' => Key4,
-        '5' => Key5,
-        '6' => Key8,
-        '7' => Key9,
-        '8' => Key6,
-        '9' => Key7,
-        _ => return None,
-    })
+/// Trait for key channel mode implementations.
+///
+/// This trait defines the interface for converting between different key channel modes
+/// and the standard Beat mode. Each mode implementation should provide methods to
+/// convert from its own format to Beat format and vice versa.
+pub trait KeyChannelMode {
+    /// Convert from this mode's format to Beat mode format.
+    ///
+    /// This method takes a (PlayerSide, Key) pair in this mode's format and converts
+    /// it to the equivalent (PlayerSide, Key) pair in Beat mode format.
+    fn to_beat(&mut self, side: PlayerSide, key: Key) -> (PlayerSide, Key);
+
+    /// Convert from Beat mode format to this mode's format.
+    ///
+    /// This method takes a (PlayerSide, Key) pair in Beat mode format and converts
+    /// it to the equivalent (PlayerSide, Key) pair in this mode's format.
+    fn map_from_beat(&mut self, side: PlayerSide, key: Key) -> (PlayerSide, Key);
```

Comment:
To avoid confusion of tuple of side and key, I'd like to treat the Beat mode format as own intermediate representation type like `pub struct BeatModeMap(pub PlayerSide, pub Key);`. Then these methods should handle it.